### PR TITLE
[IMP] bi_sql_editor manual field check performance

### PR DESCRIPTION
**Pull request by paulius-sladkevicius**
The `table_columns()` slows down modules loading significantly up to 2-3x.
In most cases `model._name.startswith(self._model_prefix)` returns `False`, so better to check it first.

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
Makes senses.

note : could you take a look on the both migration in V16 and make a review ? : 
- ``sql_request_abstract`` : https://github.com/OCA/reporting-engine/pull/669
- ``bi_sql_editor`` : https://github.com/OCA/reporting-engine/pull/670

I'll cherry-pick your commit, once merged.

In the meantime, fasttracking trivial patch.

/ocabot merge patch

**Comment by legalsylvain** [comment]
Note : pre-commit test is red. you should fix something I guess.

**Comment by paulius-sladkevicius** [comment]
@legalsylvain I think it's pre-commit issue as you can see from the error. My change is minor, shouldn't impact tests.

```  
File "/home/runner/.cache/pre-commit/repoxd7mcpcu/py_env-python3/lib/python3.11/site-packages/wrapt/decorators.py", line 34, in <module>
    from inspect import ismethod, isclass, formatargspec
ImportError: cannot import name 'formatargspec' from 'inspect' (/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/inspect.py)
```

**Comment by legalsylvain** [comment]
Any idea on that issue @OCA/community-maintainers ?

**Comment by paulius-sladkevicius** [comment]
> Any idea on that issue @OCA/community-maintainers ?

@legalsylvain as I'm not aware of this team (at least I can't see it), can you tag the specific person who could now about the pre-commit issue? Thanks.

**Comment by simahawk** [comment]
The [addons repo template](https://github.com/OCA/oca-addons-repo-template) has been updated to fix this.
To fix this repo you can:
1. checkout branch 14.0
2. run `copier -f update`
3. stage changes and commit
4. open another PR that will be fast tracked when green

Rebase here when done.

**Comment by paulius-sladkevicius** [comment]
@legalsylvain we're green.

**Comment by legalsylvain** [comment]
Thanks ! @paulius-sladkevicius. Could you take a look and make a PR against the migration in V16, I'll merge your patch.

sql_request_abstract : https://github.com/OCA/reporting-engine/pull/669
    bi_sql_editor : https://github.com/OCA/reporting-engine/pull/670

**Comment by paulius-sladkevicius** [comment]
> Thanks ! @paulius-sladkevicius. Could you take a look and make a PR against the migration in V16, I'll merge your patch.
> 
> sql_request_abstract : #669 bi_sql_editor : #670

Will do

**Comment by paulius-sladkevicius** [comment]
@legalsylvain here it goes https://github.com/legalsylvain/reporting-engine/pull/2

**Comment by paulius-sladkevicius** [comment]
@legalsylvain when it is possible to merge it?

**Pull request merged**
